kayenta-core/src/main/java/com/netflix/kayenta/canary/providers/metrics/QueryConfigUtils.java

```diff
@@ -49,49 +49,59 @@ public static String expandCustomFilter(CanaryConfig canaryConfig,
                                           CanaryMetricSetQueryConfig metricSetQuery,
                                           CanaryScope canaryScope,
                                           String[] baseScopeAttributes) throws IOException {
-    String customFilter = metricSetQuery.getCustomFilter();
+    String customInlineTemplate = metricSetQuery.getCustomInlineTemplate();
     String customFilterTemplate = metricSetQuery.getCustomFilterTemplate();
+    String templateToExpand;
+    String expandedTemplate;
 
-    log.debug("customFilter={}", customFilter);
+    log.debug("customInlineTemplate={}", customInlineTemplate);
     log.debug("customFilterTemplate={}", customFilterTemplate);
 
-    if (StringUtils.isEmpty(customFilter) && !StringUtils.isEmpty(customFilterTemplate)) {
+    if (StringUtils.isEmpty(customInlineTemplate) && StringUtils.isEmpty(customFilterTemplate)) {
+      return null;
+    }
+
+    if (!StringUtils.isEmpty(customInlineTemplate)) {
+      templateToExpand = unescapeTemplate(customInlineTemplate);
+    } else {
       Map<String, String> templates = canaryConfig.getTemplates();
 
       // TODO(duftler): Handle this as a config validation step instead.
       if (CollectionUtils.isEmpty(templates)) {
-        throw new IllegalArgumentException("Custom filter template '" + customFilterTemplate + "' was referenced, " +
-                                           "but no templates were defined.");
+        throw new IllegalArgumentException(
+            "Custom filter template '" + customFilterTemplate + "' was referenced, " +
```

Comment:
If we keep the formatting the same for this exception and the one just below, it'll cut down on the number of changed lines in the PR.